tutorial/versioning/version_ranges.rst

```diff
@@ -0,0 +1,147 @@
+.. _tutorial_versioning_version_ranges:
+
+Version ranges
+==============
+
+In the previous section, we ended with several versions of the ``pkg`` package.
+Let's remove them and create the following simple project:
+
+.. code-block:: python
+    :caption: pkg/conanfile.py
+
+    from conan import ConanFile
+
+    class pkgRecipe(ConanFile):
+        name = "pkg"
+
+.. code-block:: python
+    :caption: app/conanfile.py
+
+    from conan import ConanFile
+
+    class pkgRecipe(ConanFile):
+        name = "app"
+        requires = "pkg/1.0"
+
+Let's create ``pkg/1.0`` and install ``app``, to see it requires ``pkg/1.0``:
+
+.. code-block:: bash
+
+    $ conan remove pkg* -c
+    $ conan create pkg --version=1.0
+    ... pkg/1.0 ...
+    $ conan install app
+    ...
+    Requirements
+        pkg/1.0
+
+Then, if we create a new version of ``pkg/1.1``, it will not automatically be used by ``app``:
+
+.. code-block:: bash
+
+    $ conan create pkg --version=1.1
+    ... pkg/1.0 ...
+    # Note how this still uses the previous 1.0 version
+    $ conan install app
+    ...
+    Requirements
+        pkg/1.0
+
+So we could modify ``app`` conanfile to explictly use the new ``pkg/1.1`` version, but instead of that,
+let's use the following version-range expression (introduced by the ``[expression]`` brackets):
+
+.. code-block:: python
+    :caption: app/conanfile.py
+
+    from conan import ConanFile
+
+    class pkgRecipe(ConanFile):
+        name = "app"
+        requires = "pkg/[>=1.0 <2.0]"
+
+When we now install the dependencies of ``app``, it will automatically use the latest version in the
+range, even if we create a new one, without needing to modify the ``app`` conanfile:
+
+.. code-block:: bash
+
+    # this will now use the newer 1.1
+    $ conan install app
+    ...
+    Requirements
+        pkg/1.1
+
+    $ conan create pkg --version=1.2
+    ... pkg/1.2 ...
+    # Now it will automatically use the newest 1.2
+    $ conan install app
+    ...
+    Requirements
+        pkg/1.2
+
+This holds as long as the newer version lies within the defined range, if we create a ``pkg/2.0`` version,
+``app`` will not use it:
+
+.. code-block:: bash
+    
+    $ conan create pkg --version=2.0
+    ... pkg/2.0 ...
+    # Conan will use the latest in the range
+    $ conan install app
+    ...
+    Requirements
+        pkg/1.2
+
+
+Semantic versioning
+-------------------
+
+The semantic versioning specification or `semver <https://semver.org/>`_, specifies that packages should
+be versioned using always 3 dot-separated digits like ``MAJOR.MINOR.PATCH``, with very specific meanings for each digit.
+
+Conan extends the semver specification to any number of digits, and also allows to include letters in it.
```

Comment:
does it allow only latin letters, or any letters in general?

Reply:
[a-z][A-Z], only latin letters, yes.

Reply:
then just `[a-z]`, if only lowercase are allowed

Reply:
yes, right, only a-z